### ISSUE TITLE
Implementation of Dropout for sparse input

### ISSUE THREAD

**Issue by sqlkrad** [rusty1s/pytorch_sparse#36]
Is there any way to only dropout the non-zero elements in a spase tensor? The function can be achieved by tensorflow. I wonder if it can be done by pytorch.

https://discuss.pytorch.org/t/implementation-of-dropout-for-sparse-input/47720

**Comment by rusty1s** [comment]
Have a look at the source code from [here](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.dropout_adj).

**Comment by sqlkrad** [comment]
Thanks!

**Comment by sqlkrad** [comment]
Hi, I saw the source code of dropout_adj, I wonder how can I implement it on a sparse tensor since its parameters are LongTensor? Is there any use case?

**Comment by rusty1s** [comment]
Well, you basically just get the index and values of a sparse tensor, randomly drop some of them and put them into a new sparse tensor:
```python
index, value = adj._indices(), adj._values()
mask = value.new_full((value.size(0), ), 1 - p)
mask = torch.bernoulli(mask).to(torch.bool)
index, value = index[:, mask], value[mask]
```

**Comment by sqlkrad** [comment]
Thank you so much!

**Comment by sqlkrad** [comment]
Sorry, I tried your implementation. But it seems not modifying the **mat** in-place. Should I make a new sparse tensor by the new index and value?

```python

index, value = mat._indices(), mat._values()
mask = value.new_full((value.size(0), ), 1 - p)
mask = torch.bernoulli(mask).to(torch.bool)
index, value = index[:, mask], value[mask]
mat = torch.sparse.FloatTensor(index, value, mat.Size())
```
